packages/xdl/src/detach/AndroidShellApp.js

```diff
@@ -1111,3 +1114,102 @@ export function addDetachedConfigToExp(exp: Object, context: StandaloneContext):
   );
   return exp;
 }
+
+/**
+
+Some files in `android` directory have the following patterns of code:
+
+```
+// WHEN_PREPARING_SHELL_REMOVE_FROM_HERE
+
+// BEGIN_SDK_30
+some SDK 30-specific code
+// END_SDK_30
+
+// BEGIN_SDK_29
+some SDK 29-specific code
+// END_SDK_29
+
+...
+
+// WHEN_PREPARING_SHELL_REMOVE_TO_HERE
+```
+
+The following function replaces `BEGIN_SDK_XX` with `REMOVE_TO_HERE`
+and `END_SDK_XX` with `REMOVE_FROM_HERE`.
+
+This allows us to use `deleteLinesInFileAsync` function to remove obsolete SDKs code easily.
+
+ */
+const removeInvalidSdkLinesWhenPreparingShell = async (majorSdkVersion, filePath) => {
+  await regexFileAsync(
+    `BEGIN_SDK_${majorSdkVersion}`,
+    `WHEN_PREPARING_SHELL_REMOVE_TO_HERE`,
+    filePath
+  );
+  await regexFileAsync(
+    `END_SDK_${majorSdkVersion}`,
+    `WHEN_PREPARING_SHELL_REMOVE_FROM_HERE`,
+    filePath
+  );
+  await deleteLinesInFileAsync(
+    'WHEN_PREPARING_SHELL_REMOVE_FROM_HERE',
+    'WHEN_PREPARING_SHELL_REMOVE_TO_HERE',
+    filePath
+  );
+};
+
+async function removeObsoleteSdks(shellPath: string, requiredSdkVersion: string) {
+  const majorSdkVersion = semver.major(requiredSdkVersion);
+
```

Comment:
```
const paths = [
  // Remove obsolete `expoview-abiXX_X_X` dependencies
  path.join(shellPath, 'app', 'build.gradle'),
  // Remove obsolete `host.exp.exponent:reactandroid:XX.X.X` dependencies from expoview
  path.join(shellPath, 'expoview', 'build.gradle'),
  // ...
];
for (const x of paths) {
  await removeInvalidSdkLinesWhenPreparingShell(majorSdkVersion, x);
}
```
or you can even parallelize `removeInvalidSdkLinesWhenPreparingShell` calls by using `Promise.all`